hydrosdk/deployment_configuration.py

```diff
@@ -428,6 +430,22 @@ def with_resource_requirements(self, limits: Dict[str, str], requests: Dict[str,
             self.container_spec.resources = new_resource_requirements
         return self
 
+    def with_env(self, env: Dict[str, str]) -> 'DeploymentConfiguration':
+        """
+        Specify env for the container.
+
+        :param env: dictionary with the required environment variables.
+        """
+        if self.container_spec is None:
+            self.container_spec = ContainerSpec(env=env)
+        elif self.container_spec.env is not None:
+            raise ValueError(f"Cannot set env as {env},"
+                             f" they are already set to {self.container_spec.env}")
```

Comment:
This logic seems fine, but it contradicts with `with_pod_node_selector` behavior. These 2 methods have equal signatures `Dict[str,str] -> DeploymentConfiguration` but handle subsequent calls differently:
1. `with_env({"foo":"bar"}).with_env("bar":"foo")` raises a `ValueError`
2. `with_pod_node_selector({"foo":"bar"}).with_pod_node_selector("bar":"foo")` succeeds and updates `pod_spec.node_selector`

Personally, I'm a little bit for updating instead of raising a `ValueError` though I do not have a compelling argument.

Either way, I strongly suggest changing the behavior of either `with_env` or `with_pod_selector` so they would match.

Reply:
@KineticCookie which one seems right and why in your opinion?

Reply:
By using with_pod_nod_selector multiple times we might override some of the keys, if they happen to duplicate. That should be undesirable.